Service/Manager.php

```diff
@@ -589,15 +589,20 @@ private function parseResult($raw, $resultsType, $scrollDuration = null)
      */
     private function convertToNormalizedArray($data)
     {
-        if (array_key_exists('_source', $data)) {
-            return $data['_source'];
+        if (array_key_exists('_source', $data) && array_key_exists('_id', $data)) {
+            return [$data['_id'] => $data['_source']];
         }
 
         $output = [];
 
         if (isset($data['hits']['hits'][0]['_source'])) {
             foreach ($data['hits']['hits'] as $item) {
-                $output[] = $item['_source'];
+                $output[$item['_id']] = [
+                    '_type'  => $item['_type'],
+                    '_index'  => $item['_index'],
```

Comment:
Index will always be the same. Repository class works with 1 manager at the time.